src/inspect_ai/model/_providers/vertex.py

```diff
@@ -336,10 +336,13 @@ async def content_part(content: Content | str) -> Part:
     elif isinstance(content, ContentImage):
         image_bytes, mime_type = await file_as_data(content.image)
         return Part.from_image(image=Image.from_bytes(data=image_bytes))
+    elif isinstance(content, ContentReasoning):
+        return Part.from_text(content.reasoning or NO_CONTENT)
     else:
         if isinstance(content, ContentAudio):
             file = content.audio
-        elif isinstance(content, ContentVideo):
+        else:
+            # it's ContentVideo
```

Comment:
We _could_ future proof this by leaving the `isinstance` check for video and having a final `else: assert false`. It's basically pick one type of crash or another. OR make the the result optional and make the callers deal.